02-peer-protocol.md

```diff
@@ -49,11 +68,502 @@ transaction is confirmed are also not persistent - until you know the script
+
+#### Requirements
+
+The sender:
+  - MUST set `feerate` greater than or equal to 25/24 times the `feerate`
+    of the previously constructed transaction, rounded down.
+  - If it contributes to the transaction's funding output:
+    - MUST set `funding_output_contribution`
+
+The recipient:
+  - MUST respond either with `tx_abort` or with `tx_ack_rbf`
+  - MUST respond with `tx_abort` if:
+    - the `feerate` is not greater than or equal to 25/24 times `feerate`
+      of the last successfully constructed transaction
+  - MAY send `tx_abort` for any reason
+
+#### Rationale
+
+`feerate` is the feerate this transaction will pay. It must be at least
+1/24 greater than the last used `feerate`, rounded down to the nearest
+satoshi to ensure there is progress.
+
+E.g. if the last `feerate` was 520, the next sent `feerate` must be 541
+(520 * 25 / 24 = 541.667, rounded down to 541).
+
+If the previous transaction confirms in the middle of an RBF attempt,
+the attempt MUST be abandoned.
+
+`funding_output_contribution` is the amount of satoshis that this peer
+will contribute to the funding output of the transaction, when there is
+such an output. Note that it may be different from the contribution
+made in the previously completed transaction. If omitted, the sender is
+not contributing to the funding output.
+
+### The `tx_ack_rbf` Message
+
+1. type: 73 (`tx_ack_rbf`)
+2. data:
+   * [`channel_id`:`channel_id`]
+   * [`tx_ack_rbf_tlvs`:`tlvs`]
+
+1. `tlv_stream`: `tx_ack_rbf_tlvs`
+2. types:
+    1. type: 0 (`funding_output_contribution`)
+    2. data:
+        * [`s64`:`satoshis`]
+
+#### Requirements
+
+The sender:
+  - If it contributes to the transaction's funding output:
+    - MUST set `funding_output_contribution`
+
+The recipient:
+  - MUST respond with `tx_abort` or with a `tx_add_input` message,
+    restarting the interactive tx collaboration protocol.
+
+#### Rationale
+
+`funding_output_contribution` is the amount of satoshis that this peer
+will contribute to the funding output of the transaction, when there is
+such an output. Note that it may be different from the contribution
+made in the previously completed transaction. If omitted, the sender is
+not contributing to the funding output.
+
+It's recommended that a peer, rather than fail the RBF negotiation due to
+a large feerate change, instead stop contributing to the funding output,
+and decline to participate further in the transaction (by not contributing,
+they may obtain incoming liquidity at no cost).
+
+### The `tx_abort` Message
+
+1. type: 74 (`tx_abort`)
+2. data:
+   * [`channel_id`:`channel_id`]
+   * [`u16`:`len`]
+   * [`len*byte`:`data`]
+
+#### Requirements
+
+A sending node:
+  - MUST NOT have already transmitted `tx_signatures`
+  - SHOULD forget the current negotiation and reset their state.
+
+A receiving node:
+  - if they have already sent `tx_signatures` to the peer:
+    - MUST NOT forget the channel until any inputs to the negotiated tx
```

Comment:
I think a recommendation of double-spending the inputs could be introduced as you have the asynchronous case where Alice sends `tx_signatures` at the same time than Bob is sending `tx_abort`. Reception of `tx_abort` on Alice side puts her on a stale.

Reply:
I'm not sure what you mean? Once you've sent `tx_signatures`, you cannot forget this transaction until it has been double-spent. If you receive `tx_abort`, you should free the corresponding utxos to make sure you try double-spending it at some point.

Reply:
> If you receive tx_abort, you should free the corresponding utxos to make sure you try double-spending it at some point.

From my understanding, we’re saying the same thing namely when you receive `tx_abort` you should double-spend the utxos, though formulated differently, so sounds good.